cvat/apps/engine/views.py

```diff
@@ -879,35 +883,85 @@ def dataset_export(self, request, pk):
+        'task_name': 'segment__task__name',
+        'project_name': 'segment__task__project__name',
+        'updated_date': 'segment__task__updated_date',
+        'assignee': 'assignee__username',
+    }.get(var, var)
+
+class JobJsonLogicFilter:
+    @classmethod
+    def _build_Q(cls, rules):
+        op, args = next(iter(rules.items()))
+        if op in ['or', 'and']:
+            return reduce({
+                'or': operator.or_,
+                'and': operator.and_
+            }[op], [cls._build_Q(arg) for arg in args])
+        elif op == '!':
+            return ~cls._build_Q(args)
+        elif op == '!!':
+            return cls._build_Q(args)
+        elif op == 'var':
+            return Q(**{args + '__isnull': False})
+        elif op in ['==', '<', '>', '<=', '>='] and len(args) == 2:
+            var = get_job_viewset_lookup_field(args[0]['var'])
+            q_var = var + {
+                '==': '',
+                '<': '__lt',
+                '<=': '__lte',
+                '>': '__gt',
+                '>=': '__gte'
+            }[op]
+            return Q(**{q_var: args[1]})
+        elif op == 'in':
+            if isinstance(args[0], dict):
+                var = get_job_viewset_lookup_field(args[0]['var'])
+                return Q(**{var + '__in': args[1]})
+            else:
+                var = get_job_viewset_lookup_field(args[1]['var'])
+                return Q(**{var + '__contains': args[0]})
+        elif op == '<=' and len(args) == 3:
+            var = get_job_viewset_lookup_field(args[1]['var'])
+            return Q(**{var + '__gte': args[0]}) & Q(**{var + '__lte': args[2]})
+        else:
+            raise NotImplementedError(f'{op} operation with {args} arguments is not implemented')
 
-    class Meta:
-        model = Job
-        fields = ("assignee", )
+    @classmethod
+    def filter(cls, queryset, rules):
+        return queryset.filter(cls._build_Q(rules))
 
 @method_decorator(name='retrieve', decorator=swagger_auto_schema(operation_summary='Method returns details of a job'))
 @method_decorator(name='update', decorator=swagger_auto_schema(operation_summary='Method updates a job by id'))
 @method_decorator(name='partial_update', decorator=swagger_auto_schema(
     operation_summary='Methods does a partial update of chosen fields in a job'))
 class JobViewSet(viewsets.GenericViewSet, mixins.ListModelMixin,
     mixins.RetrieveModelMixin, mixins.UpdateModelMixin):
-    queryset = Job.objects.all().order_by('id')
-    filterset_class = JobFilter
+    queryset = Job.objects.all()
     iam_organization_field = 'segment__task__organization'
 
     def get_queryset(self):
         queryset = super().get_queryset()
+
         if self.action == 'list':
             perm = JobPermission.create_list(self.request)
             queryset = perm.filter(queryset)
 
-        return queryset
+        order_by = self.request.query_params.get('order_by', 'id').split(',')
```

Comment:
@bsekachev , let me look at the functionality. I believe we should have 3 separate ways for filters as described here: https://www.django-rest-framework.org/api-guide/filtering/#api-guide

- filter - a json complex filter condition
- sort - to sort by multiple fields
- search - simple search like `assignee:word ...`

I will prefer to remove the functionality to search by a field like `assignee=nmano` directly using query parameters. Probably the functionality can be included into `search` using special syntax.

Reply:
> I will prefer to remove the functionality to search by a field like assignee=nmano directly using query parameters. Probably the functionality can be included into search using special syntax

Totally agree :)